### ISSUE TITLE
How to use shared AppGroup UserDefaults in macOS and Xcode extension

### ISSUE THREAD

**Issue by onmyway133** [onmyway133/blog#201]
- Go to both app and extension target, under Capabilities, enable AppGroup
- Specify `$(TeamIdentifierPrefix)group.com.onmyway133.MyApp`
- `$(TeamIdentifierPrefix)` will expand to something like `T78DK947F3.`, with `.`

- Then using is like a normal UserDefaults

```swift
let defaults = UserDefaults(suiteName: "T78DK947F3.group.com.onmyway133.MyApp")

defaults?.set(true, forKey: "showOptions")
defaults?.synchronize()
```

**Comment by radianttap** [comment]
It's interesting that TeamIdentifierPrefix is needed (on iOS it is not between app and extensions).

**Comment by onmyway133** [comment]
@radianttap ja, luckily Xcode is helpful. It prepopulates with `TeamIdentifierPrefix` and we can preview its value

**Comment by harryworld** [comment]
`$(TeamIdentifierPrefix)` will generate as something like `group.T78DK947F3.` in Xcdoe 11.3. Do you know if this will pass App Store submission?

**Comment by AkdM** [comment]
Do you know if there's a way to get the TeamIdentifierPrefix in code instead of typing the identifier by hand? Also, same question as @harryworld, would it pass App Store submission if typed manually?

**Comment by avioli** [comment]
@AkdM Add an entry to your `Info.plist` file:

```xml
<key>TeamIdentifierPrefix</key>
<string>$(DEVELOPMENT_TEAM).</string>
```

NOTE: I guess you know, but still - you can name it whatever you want and you don't have to add the dot at the end of the value if you don't want to.

Then read the value at runtime:

```swift
class SharedPersistantContainer: NSPersistentContainer {
    static let teamIdPrefix = Bundle.main.object(forInfoDictionaryKey: "TeamIdentifierPrefix")!
    static let sharedUrl = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: "\(teamIdPrefix)group.com.akdm.sharedData")!
    let storeDescription = NSPersistentStoreDescription(url: sharedUrl)
    
    override class func defaultDirectoryURL() -> URL {
        return sharedUrl
    }
}
```

Which would give you something like this on macOS:
`file:///Users/akdm/Library/Group%20Containers/T2FP2A1KA4.group.com.akdm.sharedData/`

**Comment by kravchuk-se** [comment]
Hey, guys. I'm facing the same issue in my project. Xcode 12.1 doesn't allow me to start a group identifier with the team id. I couldn't find any information about that besides this thread. The project is working fine without the team id, but I am also afraid it will not pass submission to App Store. Do you have any information about that?
